### ISSUE TITLE
Is x-correlator support required at OIDC endpoints?

### ISSUE THREAD

**Issue by trehman-gsma** [camaraproject/IdentityAndConsentManagement#240]
_**Context**_
The CAMARA APIs define `401` and `403` error responses that include an optional [`x-correlator`](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#9-architecture-headers) header. For example, in [SimSwap v1.0.0](https://github.com/camaraproject/SimSwap/blob/r1.3/code/API_definitions/sim-swap.yaml#L281)  and [CAMARA_common.yaml r0.4.0](https://github.com/camaraproject/Commonalities/blob/r0.4.0/artifacts/CAMARA_common.yaml#L163). In architectures where the AZ validates the request before it reaches the service API, `x-correlator` support may be required at the AZ level.

_**Questions**_
* Is x-correlator support required at all OIDC endpoints (e.g. `GET /auth`,  `POST /bc-authorize`, `POST /token`)?
* Should the ICM documentation make a note about expected `x-correlator` behaviour for `401`/`403` service API responses?

**Comment by jpengar** [comment]
CAMARA defines the x-correlator header for CAMARA APIs calls as per https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#9-architecture-headers. 

However, the authentication flows follow the OAuth/OIDC/CIBA standards and have their own interface definitions, errors, headers, etc. I see no reason for ICM to require x-correlator header for the authentication flows endpoints just because it is defined for CAMARA APIs requests/responses.

**Comment by AxelNennker** [comment]
I agree with @jpengar adding a header makes the Authorization Server proprietary and we should avoid that.

@trehman-gsma please close this. If you accept our answers.

**Comment by jpengar** [comment]
@trehman-gsma Can we close this issue then?

**Comment by trehman-gsma** [comment]
Hi @jpengar @AxelNennker. Thanks for your response and sorry for the late reply. I will close this issue.
